docs/add_a_validation.md

```diff
@@ -0,0 +1,295 @@
+                    ->addModelTransformer(new StringToArrayTransformer())
+                    // Deactivate auto-initialize as we're not on the root form.
+                    ->setAutoInitialize(false)->getForm()
+            );
+    }
+}
+```
+
+Next thing is to register the FormMapper as a service, so the system would know to use it. To achieve this, let's add the following lines to `fieldtypes.yml`:
+```yml
+// Resources/config/fieldtypes.yml
+
+    ezsystems.tweetbundle.fieldtype.eztweet.form_mapper:
+        class: EzSystems\TweetFieldTypeBundle\eZ\Publish\FieldType\Tweet\FormMapper
+        tags:
+            - {name: ez.fieldFormMapper.definition, fieldType: eztweet}
+```
+
+## Add field definition edit view
+
+We have the new part of the form defined, but we still need to show it to the user. To do that, we need to create file containing the view:
+```html
+// Resources/views/platformui/content_type/edit/eztweet.html.twig
+
+{% block eztweet_field_definition_edit %}
+    <div class="eztweet-validator author_list{% if group_class is not empty %} {{ group_class }}{% endif %}">
+        {{- form_label(form.authorList) -}}
+        {{- form_errors(form.authorList) -}}
+        {{- form_widget(form.authorList) -}}
+    </div>
+{% endblock %}
+```
+
+Also, we need to register it by editing the file `ez_field_templates.yml`:
+```yml
+// Resources/config/ez_field_templates.yml
+
+        fielddefinition_edit_templates:
+            - {template: EzSystemsTweetFieldTypeBundle:platformui/content_type/edit:eztweet.html.twig, priority: 0}
+```
+
+## Implement `toStorageFieldDefinition()` and `toFieldDefinition()` methods in LegacyConverter
+
+The last thing to do is to make sure that validation data is properly saved into and retrieved from the database. To achieve this, we need to implement these two functions in LegacyConverter file:
+```php
+// eZ/Publish/FieldType/Tweet/LegacyConverter.php
+
+public function toStorageFieldDefinition(FieldDefinition $fieldDef, StorageFieldDefinition $storageDef)
+{
+    $storageDef->dataText1 = json_encode(
+        $fieldDef->fieldTypeConstraints->validators['TweetValueValidator']['authorList']
+    );
+}
+
+public function toFieldDefinition(StorageFieldDefinition $storageDef, FieldDefinition $fieldDef)
+{
+    $authorList = json_decode($storageDef->dataText1);
+    if (!empty($authorList)) {
+        $fieldDef->fieldTypeConstraints->validators = [
+            'TweetValueValidator' => [
+                'authorList' => $authorList
+            ],
+        ];
+    }
+}
+```
+
+`toStorageFieldDefinition()` converts a field definition to a legacy one using the `dataText1` field (which in this example means converting it to json), and `toFieldDefinition()` converts a stored legacy field definition to an API field definition (which means converting it back to an array according to validation schema).
+
+You should now be able to configure list of authors when editing Content Type with Tweet Field Type.
```

Comment:
You should now be able to configure **a** list of authors when editing **a** Content Type with **the** Tweet Field Type.